model/Orders.php

```diff
@@ -0,0 +1,317 @@
+<?php
+
+class OrdersModel {
+
+    private $database;
+
+
+    /**
+     * OrdersModel constructor.
+     * 
+     * @param PDO $database The database connection.
+     */
+    public function __construct($database) {
+        $this->database = $database;
+
+    }
+
+    /**
+     * Retrieve Order details by OrderID.
+     * 
+     * @param int $orderID The unique identifier of the Order.
+     * @return array|null The Order details or null if not found.
+     */
+    public function getOrderByID($orderID) {
+        $query = "SELECT * FROM `Orders` WHERE `OrderID` = :orderID";
+        $statement = $this->database->prepare($query);
+        $statement->bindParam(':orderID', $orderID, PDO::PARAM_INT);
+
+        if ($statement->execute()) {
+            $order = $statement->fetch(PDO::FETCH_ASSOC);
+            return $order ? $order : null;
+        } else {
+            return null; // Failed to execute query
+        }
+    }
+
+
+    /**
+     * Retrieve all Orders for a given Customer.
+     * 
+     * @param int $customerID The unique identifier of the Customer.
+     * @return array|null The Orders or null if not found.
+     */
+    public function getAllOrdersByCustomerID($customerID) {
+        $query = "SELECT * FROM `Orders` WHERE `CustomerID` = :customerID";
+        $statement = $this->database->prepare($query);
+        $statement->bindParam(':customerID', $customerID, PDO::PARAM_INT);
+
+        if ($statement->execute()) {
+            $orders = $statement->fetchAll(PDO::FETCH_ASSOC);
+            return $orders ? $orders : null;
+        } else {
+            return null; // Failed to execute query
+        }
+    }
+
+
+    /**
+     * Retreive OrderStatusID by Name from OrderStatus table.
+     * 
+     * @param string $orderStatusName The name of the OrderStatus.
+     * @return int|null The OrderStatusID or null if not found.
+     */
+    public function getOrderStatusIDByName($orderStatusName) {
+        $query = "SELECT `OrderStatusID` FROM `OrderStatus` WHERE `Name` = :orderStatusName";
```

Comment:
Won't this return every orderStatus with the same name (i.e. returning multiple baskets etc.)?

Reply:
Yes, however in the evotechDB.sql perhaps I can set it up so the the status name is unique?
As the main orders table stores the status as an int (to reduce the amount of time a string like "basket' is stored in the DB), this function is used to gain the associated ID for a given string.
![image](https://github.com/aston-cs2tp-evotech/evotech/assets/63474858/b6846334-32ed-44b5-89ce-a90182febee9)